pkg/security/security_profile/profile/manager.go

```diff
@@ -412,6 +412,8 @@ func (m *SecurityProfileManager) OnCGroupDeletedEvent(workload *cgroupModel.Cach
 func (m *SecurityProfileManager) ShouldDeleteProfile(profile *SecurityProfile) {
 	m.profilesLock.Lock()
 	defer m.profilesLock.Unlock()
+	m.pendingCacheLock.Lock()
```

Comment:
Could we not just use one big lock for both use case ?

Reply:
Currently there are multiple combinations of the locks (profile, profilesLock, pendingCacheLock), I'm not sure we can easily use one big lock with the current state of the code.

Reply:
Ok !